pkg/scheduler/api/podgroup_info/allocation_info.go

```diff
@@ -24,6 +24,15 @@ func HasTasksToAllocate(podGroupInfo *PodGroupInfo, isRealAllocation bool) bool
 	return false
 }
 
+func HasTasksAllocated(podGroupInfo *PodGroupInfo) bool {
```

Comment:
Instead, take a look at `PodGroupInfo.GetNumPendingTasks` or `PodGroupInfo.GetNumGatedTasks` at `pkg/scheduler/api/podgroup_info/job_info.go`. They use `podGroupInfo.PodStatusIndex` which is simpler

Reply:
PodStatusIndex stores only per-status. Is it safe enough to just look at if all tasks are in pending?
I wanted to flag any status that might indicate the Job has already gotten a LastStartTimestamp, which I suspected would contain both Allocated, Bound, Binding etc (hence using the AllocatedStatus helper fn).